src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,131 @@
+# lm-buddy entrypoint to run evaluations using a Prometheus inference server
+# see https://github.com/kaistAI/prometheus/blob/main/evaluation/benchmark/run_absolute_scoring.py
+
+from lm_buddy.jobs.configs import PrometheusJobConfig
+from lm_buddy.integrations.huggingface import HuggingFaceAssetLoader
+from lm_buddy.integrations.wandb import (
+    ArtifactType, 
+    ArtifactLoader, 
+    build_file_artifact, 
+    wandb_init_from_config,
+)
+from fastchat.conversation import get_conv_template
+from transformers import AutoTokenizer
+from openai import OpenAIError, OpenAI
+
+from tqdm import tqdm
+import os
+import json
+import copy
+
+class BadResponseException(Exception):
+    def __init__(self, message, error):
+        self.message = message
+        self.error = error
+
+
+def openai_completion(config, client, prompt):
+    return client.completions.create(
+        model = "kaist-ai/prometheus-13b-v1.0",
+        prompt = prompt,
+        best_of = config.prometheus.best_of,
+        max_tokens = config.prometheus.max_tokens,
+        frequency_penalty = config.prometheus.frequency_penalty,
+        temperature = config.prometheus.temperature,
+        top_p = config.prometheus.top_p
+    )
+
+
+def parse_response(response):
+    try:
+        assert response is not None
+        response_text = response.choices[0].text
+        feedback, score = response_text.split('[RESULT]')
+        feedback = feedback.strip()
+        score = score.strip()
+        assert score in ["1","2","3","4","5"]
+    except (ValueError, AssertionError) as e:
+        raise BadResponseException("Server returned a bad response", e)
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction):
+    conv = get_conv_template("llama-2")
+    conv.set_system_message("You are a fair evaluator language model.")
+    conv.append_message(conv.roles[0], instruction)
+    conv.append_message(conv.roles[1], None)
+    return conv.get_prompt()
+
+
+def run_prometheus(config: PrometheusJobConfig, artifact_loader: ArtifactLoader):
+
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    artifact_path,_ = hf_loader.resolve_asset_path(config.dataset.load_from)
+    dataset_fname = os.path.join(artifact_path, config.dataset.load_from.name)
```

Comment:
Oh yes, I kinda misused the class because I did not want to create another one specifically for JSON files 😅  (bad, I know, but I think we should expect this kind of behavior from lm-buddy users and either embrace the "there's more than one way to do it" paradigm or provide more specific classes, e.g. HFTextDatasetConfig for HF datasets and TextDatasetConfig for other file types).
For the dataset specifically, I agree. I like that it works on those original JSON files though. I'd like to make a new entrypoint specifically for preprocessing that will take the JSON files and save them as HF dataset artifacts (perhaps in next PR). This can also be a good example to add to our documentation too!

Reply:
I think for now we should rely on the existing asset load mechanism for HF datasets, since the configs and associated methods were geared towards that. I think it would be simplify this code alot. 

But I agree in general that supporting other dataset types could be a long term goal, and may require a more holistic refactor of the dataset loading functions.

>I'd like to make a new entrypoint specifically for preprocessing that will take the JSON files and save them as HF dataset artifacts (perhaps in next PR).

Sure! We can add some helper functions here, and perhaps add a section on that to the `dataset_preprocessing` example notebook that already exists. The HuggingFace `Dataset` class contains some extra constructors methods for instantiating their class from various data formats, I would check that out.

Reply:
Added load/save of datasets in HF format!
The output is a file that is updated incrementally (you can just `tail` it to follow its progress and if the process breaks you have not lost everything), and at the end it is synced with wandb if tracking is enabled.
The code to load the json file and save it as HF dataset is literally two lines but I am happy to add it to documentation in a follow-up!